yt/fields/tests/test_ambiguous_fields.py

```diff
@@ -11,34 +8,29 @@ def test_ambiguous_fails():
     ds = fake_amr_ds(particles=10)
     msg = "The requested field name '{}' is ambiguous"
 
-    fnames_to_ftype = defaultdict(list)
```

Comment:
Why are all of these tests disappearing? From reading your code below, it seems you are testing that accessing ` mock_field` raises a warning, but you are not verifying accessing `("io", "mock_field")` and `("gas", "mock_field")` are fine.

Reply:
I'd sleep tighter if you also checked that `ds.r["io", "mock_field"]` and `ds.r["gas", "mock_field"]` worked without raising any warning.

Reply:
Done !